.cicd/scripts/srw_test.sh

```diff
@@ -17,5 +17,190 @@ else
     workspace="$(cd -- "${script_dir}/../.." && pwd)"
 fi
 
+# Normalize Parallel Works cluster platform value.
+declare platform
+if [[ "${SRW_PLATFORM}" =~ ^(az|g|p)cluster_noaa ]]; then
+    platform='noaacloud'
+else
+    platform="${SRW_PLATFORM}"
+fi
+
+declare we2e_experiment_base_dir
+if [[ -n "${SRW_WE2E_EXPERIMENT_BASE_DIR}" ]]; then
+    we2e_experiment_base_dir="${SRW_WE2E_EXPERIMENT_BASE_DIR}"
+else
+    we2e_experiment_base_dir="${workspace}/experiments"
+fi
+
+we2e_test_dir="${workspace}/regional_workflow/tests/WE2E"
+
+we2e_test_file="${we2e_test_dir}/experiments.txt"
+
+# The fundamental set of end-to-end tests to run.
+declare -a we2e_fundamental_tests
+we2e_fundamental_tests=('grid_RRFS_CONUS_25km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_v16'
```

Comment:
I think there are more fundamental tests than these.  Same for comprehensive tests.  Will comment more soon.

Reply:
@jessemcfarland Here's the latest set of fundamental tests:
```
grid_RRFS_CONUS_25km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_v16
grid_RRFS_CONUS_25km_ics_FV3GFS_lbcs_FV3GFS_suite_RRFS_v1beta
grid_RRFS_CONUS_25km_ics_FV3GFS_lbcs_RAP_suite_HRRR
grid_RRFS_CONUS_25km_ics_NAM_lbcs_NAM_suite_HRRR
grid_RRFS_CONUS_25km_ics_NAM_lbcs_NAM_suite_RRFS_v1beta
grid_RRFS_CONUScompact_25km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_v16
grid_RRFS_CONUScompact_25km_ics_HRRR_lbcs_HRRR_suite_HRRR
grid_RRFS_CONUScompact_25km_ics_HRRR_lbcs_RAP_suite_HRRR
grid_RRFS_CONUScompact_25km_ics_HRRR_lbcs_RAP_suite_RRFS_v1beta
grid_SUBCONUS_Ind_3km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_v16
grid_SUBCONUS_Ind_3km_ics_HRRR_lbcs_RAP_suite_HRRR
grid_SUBCONUS_Ind_3km_ics_HRRR_lbcs_RAP_suite_RRFS_v1beta
nco_grid_RRFS_CONUScompact_25km_ics_HRRR_lbcs_RAP_suite_HRRR
community_ensemble_2mems
custom_ESGgrid
deactivate_tasks
inline_post
MET_ensemble_verification
MET_verification
nco_ensemble
pregen_grid_orog_sfc_climo
specify_DOT_OR_USCORE
specify_DT_ATMOS_LAYOUT_XY_BLOCKSIZE
specify_RESTART_INTERVAL
specify_template_filenames
```
Also, you can remove `subhourly_post` and `subhourly_post_ensemble_2mems` from the comprehensive tests because I think the subhourly post feature (really should be called "subhourly forecast output" or something similar) is broken.  

Please go ahead and update the list of tests here.  I know I said it during the meeting, but I'll mention again here for the record that @mkavulich is going to add a variable to the WE2E test configuration files that indicates whether a given test is part of the fundamental list, and he will add a flag to `run_WE2E_tests.sh` that runs just the fundamental set of tests.  Once that feature is in, you won't have to maintain a list in this script.

Finally, I just checked the `run_WE2E_tests.sh` script, and I don't see any if-statements, etc, that remove tests that shouldn't be run on certain platforms, e.g. removing `get_from_HPSS_...` on platforms that don't have access to NOAA HPSS.  I remember @mkavulich mentioned he wanted to add that to the script (so you won't have to do it in this script and the capability will be available to users who run tests from the command line), but I don't know where that stands.

Reply:
@gsketefian I updated the fundamental tests lists (and subsequently the comprehensive tests lists). **NOTE**: I have logic in the script that adds the `MET_ensemble_verification`, `MET_verification`, and `pregen_grid_orog_sfc_climo` tests only when the platform is NOT Gaea or one of the Parallel Works clusters. Is this correct?

Can you verify that the comprehensive tests list is correct as well? Is there a definitive list of all tests somewhere in the repository?

Reply:
Hi @jessemcfarland, I just checked on Gaea, and it looks like METplus is not installed there (and the MET version is pretty old; it's v8.1, but latest is v10.1.1), so you're right, the `MET_ensemble_verification` and `MET_verification` tests cannot be run there.  For `pregen_grid_orog_sfc_climo`, I cannot tell if the files needed to run that test are in the staging directory because the staging directory does not have read permission for "other" (so can't see contents).  I assume you tried running `pregen_grid_orog_sfc_climo` and it failed, is that right?

For now, let's do what you're doing and remove these 3 tests on `gaea` and `noaacloud`, but we should discuss this at the next code management meeting.  I've put some points to discuss in the meeting [notes](https://docs.google.com/document/d/1HS7u5xo9VoS0Xh2BR4gMUXKgtYkwXJtHWllYotEeiFQ/edit) (feel free to raise other concerns there).  One outstanding question is what to do with tests that work on some platforms but not others.  Do we still run the test so we're reminded that there's a failure that needs to be fixed?  Removing them here in this script is a bit sneaky in that the user may not realize it's happening and will assume all tests passed.

For the comprehensive tests, the idea is that we don't need a list and should run all available tests.  So eventually (after @mkavulich is done with his changes), we shouldn't need a list here.  For now, below is a list that you can use for the `develop` branch (I think the one you have was for the last release branch).  I've removed `subhourly_post` and `subhourly_post_ensemble_2mems` from the list as well as a couple of others that are actually alternate names (these are `nco_inline_post` and `template_vars`, which are just symlinks to `nco_grid_RRFS_CONUScompact_25km_ics_HRRR_lbcs_RAP_suite_HRRR` and `deactivate_tasks`, respectively).
```
MET_ensemble_verification
MET_verification
community_ensemble_008mems
community_ensemble_2mems
custom_ESGgrid
custom_GFDLgrid
custom_GFDLgrid__GFDLgrid_USE_NUM_CELLS_IN_FILENAMES_eq_FALSE
custom_GFDLgrid__GFDLgrid_USE_NUM_CELLS_IN_FILENAMES_eq_TRUE
deactivate_tasks
get_from_HPSS_ics_FV3GFS_lbcs_FV3GFS_fmt_grib2_2019061200
get_from_HPSS_ics_FV3GFS_lbcs_FV3GFS_fmt_grib2_2019101818
get_from_HPSS_ics_FV3GFS_lbcs_FV3GFS_fmt_grib2_2020022518
get_from_HPSS_ics_FV3GFS_lbcs_FV3GFS_fmt_grib2_2020022600
get_from_HPSS_ics_FV3GFS_lbcs_FV3GFS_fmt_grib2_2021010100
get_from_HPSS_ics_FV3GFS_lbcs_FV3GFS_fmt_nemsio
get_from_HPSS_ics_FV3GFS_lbcs_FV3GFS_fmt_nemsio_2019061200
get_from_HPSS_ics_FV3GFS_lbcs_FV3GFS_fmt_nemsio_2019101818
get_from_HPSS_ics_FV3GFS_lbcs_FV3GFS_fmt_nemsio_2020022518
get_from_HPSS_ics_FV3GFS_lbcs_FV3GFS_fmt_nemsio_2020022600
get_from_HPSS_ics_FV3GFS_lbcs_FV3GFS_fmt_nemsio_2021010100
get_from_HPSS_ics_FV3GFS_lbcs_FV3GFS_fmt_netcdf_2021062000
get_from_HPSS_ics_FV3GFS_lbcs_FV3GFS_fmt_netcdf_2022060112_48h
get_from_HPSS_ics_GSMGFS_lbcs_GSMGFS
get_from_HPSS_ics_HRRR_lbcs_RAP
get_from_HPSS_ics_RAP_lbcs_RAP
get_from_NOMADS_ics_FV3GFS_lbcs_FV3GFS_fmt_nemsio
inline_post
nco_ensemble
pregen_grid_orog_sfc_climo
specify_DOT_OR_USCORE
specify_DT_ATMOS_LAYOUT_XY_BLOCKSIZE
specify_EXTRN_MDL_SYSBASEDIR_ICS_LBCS
specify_RESTART_INTERVAL
specify_template_filenames
grid_CONUS_25km_GFDLgrid_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_v16
grid_CONUS_3km_GFDLgrid_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_v16
grid_RRFS_AK_13km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_v16
grid_RRFS_AK_3km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_v16
grid_RRFS_CONUS_13km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_v15p2
grid_RRFS_CONUS_13km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_v16
grid_RRFS_CONUS_13km_ics_FV3GFS_lbcs_FV3GFS_suite_HRRR
grid_RRFS_CONUS_13km_ics_FV3GFS_lbcs_FV3GFS_suite_RRFS_v1beta
grid_RRFS_CONUS_25km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_2017_gfdlmp
grid_RRFS_CONUS_25km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_2017_gfdlmp_regional
grid_RRFS_CONUS_25km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_v15p2
grid_RRFS_CONUS_25km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_v16
grid_RRFS_CONUS_25km_ics_FV3GFS_lbcs_FV3GFS_suite_HRRR
grid_RRFS_CONUS_25km_ics_FV3GFS_lbcs_FV3GFS_suite_RRFS_v1beta
grid_RRFS_CONUS_25km_ics_FV3GFS_lbcs_RAP_suite_HRRR
grid_RRFS_CONUS_25km_ics_GSMGFS_lbcs_GSMGFS_suite_GFS_2017_gfdlmp
grid_RRFS_CONUS_25km_ics_GSMGFS_lbcs_GSMGFS_suite_GFS_v15p2
grid_RRFS_CONUS_25km_ics_GSMGFS_lbcs_GSMGFS_suite_GFS_v16
grid_RRFS_CONUS_25km_ics_NAM_lbcs_NAM_suite_HRRR
grid_RRFS_CONUS_25km_ics_NAM_lbcs_NAM_suite_RRFS_v1beta
grid_RRFS_CONUS_3km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_v15_thompson_mynn_lam3km
grid_RRFS_CONUS_3km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_v15p2
grid_RRFS_CONUS_3km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_v16
grid_RRFS_CONUS_3km_ics_FV3GFS_lbcs_FV3GFS_suite_HRRR
grid_RRFS_CONUS_3km_ics_FV3GFS_lbcs_FV3GFS_suite_RRFS_v1beta
grid_RRFS_CONUScompact_13km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_v16
grid_RRFS_CONUScompact_13km_ics_HRRR_lbcs_RAP_suite_HRRR
grid_RRFS_CONUScompact_13km_ics_HRRR_lbcs_RAP_suite_RRFS_v1beta
grid_RRFS_CONUScompact_25km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_v16
grid_RRFS_CONUScompact_25km_ics_HRRR_lbcs_HRRR_suite_HRRR
grid_RRFS_CONUScompact_25km_ics_HRRR_lbcs_HRRR_suite_RRFS_v1beta
grid_RRFS_CONUScompact_25km_ics_HRRR_lbcs_RAP_suite_HRRR
grid_RRFS_CONUScompact_25km_ics_HRRR_lbcs_RAP_suite_RRFS_v1beta
grid_RRFS_CONUScompact_3km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_v16
grid_RRFS_CONUScompact_3km_ics_HRRR_lbcs_RAP_suite_GFS_v15p2
grid_RRFS_CONUScompact_3km_ics_HRRR_lbcs_RAP_suite_HRRR
grid_RRFS_CONUScompact_3km_ics_HRRR_lbcs_RAP_suite_RRFS_v1beta
grid_RRFS_NA_13km_ics_FV3GFS_lbcs_FV3GFS_suite_RRFS_v1beta
grid_RRFS_NA_3km_ics_FV3GFS_lbcs_FV3GFS_suite_RRFS_v1beta
grid_RRFS_SUBCONUS_3km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_v16
grid_RRFS_SUBCONUS_3km_ics_HRRR_lbcs_RAP_suite_GFS_v15p2
grid_SUBCONUS_Ind_3km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_v16
grid_SUBCONUS_Ind_3km_ics_HRRR_lbcs_RAP_suite_HRRR
grid_SUBCONUS_Ind_3km_ics_HRRR_lbcs_RAP_suite_RRFS_v1beta
nco_grid_RRFS_CONUS_13km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_v16
nco_grid_RRFS_CONUS_3km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_v15_thompson_mynn_lam3km
nco_grid_RRFS_CONUScompact_25km_ics_HRRR_lbcs_RAP_suite_HRRR
```